src/main/kotlin/net/maizegenetics/phgv2/api/HaplotypeGraph.kt

```diff
@@ -513,4 +516,40 @@ class HaplotypeGraph(hvcfFiles: List<String>) {
         return result
     }
 
+    /**
+     * Returns the checksum for this graph.
+     * The checksum is based on the sample names, ranges, and haplotypes.
+     * This is used to determine if two graphs are the same.
+     */
+    private fun checksum(): String {
+
+        val digester = MessageDigest.getInstance("MD5")
+
+        sampleNames.forEach { sampleName ->
+            digester.update(sampleName.toByteArray())
+        }
+
+        ranges().forEach { range ->
+            digester.update(range.toString().toByteArray())
+        }
+
+        rangeByGameteIdToHapid.forEach { range ->
+            range.forEach { sample ->
+                sample.forEach { hapid ->
+                    digester.update(hapid.toByteArray())
+                }
+            }
+        }
+
+        val bytes = digester.digest()
+
+        // convert the byte to hex format
+        val builder = StringBuilder()
+        for (i in bytes.indices) {
```

Comment:
Humor me and change the "i" to "idx" in "i in bytes.indices".

Reply:
I changed it